src/charm.py

```diff
@@ -303,19 +247,71 @@ def _remove_repo_folder(self):
         # but to keep unittest simpler, doing it from the charm container's mount point
         shutil.rmtree(self._repo_path, ignore_errors=True)
 
+    def _command(self, one_time=False) -> List[str]:
+        """Construct the command line for running git-sync.
+
+        Args:
+            one_time: flag for adding the `--one-time` argument to have git-sync exit after the
+            first sync.
+        """
+        repo = cast(str, self.config.get("git_repo"))
+        branch = cast(str, self.config.get("git_branch"))
+        rev = cast(str, self.config.get("git_rev"))
+        wait = str(self.config.get("git_wait"))  # converting to str so that 0 evaluates as True
+
+        cmd = ["/git-sync"]
+        cmd.extend(["--repo", repo])
+        if branch:
+            cmd.extend(["--branch", branch])
+        if rev:
+            cmd.extend(["--rev", rev])
+        cmd.extend(
+            [
+                "--depth",
+                "1",
```

Comment:
why a depth of one? this requires explaining

Reply:
There is already an issue for that :)
https://github.com/canonical/cos-configuration-k8s-operator/issues/7